From 05cbb7a26a764a3c1ef3e3be61213a5fd740c6a1 Mon Sep 17 00:00:00 2001
From: Martins Mozeiko <martins.mozeiko@lge.com>
Date: Fri, 13 Jul 2018 16:45:52 -0700
Subject: [PATCH] fix new/delete mismatch

Memory allocated with new must be freed with delete, not free.
---
 src/executor.cpp     | 2 +-
 src/rcl_bindings.cpp | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/executor.cpp b/src/executor.cpp
index b88d632..b1fc08e 100644
--- a/src/executor.cpp
+++ b/src/executor.cpp
@@ -62,7 +62,7 @@ void Executor::Stop() {
                  // Important Notice:
                  //  This might be called after Executor::~Executor()
                  //  Don't free Executor::async_ in Executor's dtor
-                 free(async);
+                 delete async;
                  handle_closed = true;
                });
       while (!handle_closed)
diff --git a/src/rcl_bindings.cpp b/src/rcl_bindings.cpp
index 25c462e..3f90060 100644
--- a/src/rcl_bindings.cpp
+++ b/src/rcl_bindings.cpp
@@ -669,8 +669,7 @@ const rmw_qos_profile_t* GetQosProfileFromObject(v8::Local<v8::Object> object) {
 }
 
 rmw_qos_profile_t* GetQoSProfile(v8::Local<v8::Value> qos) {
-  rmw_qos_profile_t* qos_profile =
-      reinterpret_cast<rmw_qos_profile_t*>(malloc(sizeof(rmw_qos_profile_t)));
+  rmw_qos_profile_t* qos_profile = new rmw_qos_profile_t();
 
   if (qos->IsString()) {
     *qos_profile = *GetQoSProfileFromString(
